pilot/pkg/serviceregistry/kube/controller/serviceexportcache.go

```diff
@@ -58,8 +58,11 @@ func newServiceExportCache(c *Controller) serviceExportCache {
 		_ = dInformer.Informer().SetTransform(kube.StripUnusedFields)
 		ec := &serviceExportCacheImpl{
 			Controller: c,
-			informer:   dInformer.Informer(),
-			lister:     dInformer.Lister(),
+		}
+		if c.opts.DiscoveryNamespacesFilter != nil {
+			ec.filteredInformer = informer.NewFilteredSharedIndexInformer(c.opts.DiscoveryNamespacesFilter.Filter, dInformer.Informer())
+		} else {
+			ec.filteredInformer = informer.NewFilteredSharedIndexInformer(nil, dInformer.Informer())
```

Comment:
same comment about reducing `if`

Reply:
c.opts.DiscoveryNamespacesFilter can be nil, this can be optimized later, because we need to filter namespace string as well